docs/standard/native-interop/best-practices.md

```diff
@@ -251,6 +251,56 @@ finally
 }
 ```
 
+## Windows and non-Windows data type considerations
+
+There are types in the C language that have latitude in how they are defined. When writing cross-platform interop, cases can arise where platforms differ and can cause issues if not considered.
+
+### C `long`
+
+The `long` keyword in C is defined to have ["at least 32"](https://en.cppreference.com/w/c/language/arithmetic_types) bits. This means there is a minimum required but platforms are not limited to that. The table below illustrates the differences in provided bits for the C `long` data type between platforms.
+
+|             | 32-bit | 64-bit |
+|:------------|:-------|:-------|
+| Windows     | 32     | 32     |
+| macOS/\*nix | 32     | 64     |
+
+These differences make authoring cross-platform P/Invokes difficult when the native function is defined to use `long` on all platforms. When targeting .NET 5 and prior .NET versions, you should declare separate Windows and non-Windows signatures to handle the problem.
+
+```csharp
+static readonly bool IsWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+// Cross platform C function
+// void Function(long a);
+
+[DllImport("NativeLib", EntryPoint = "Function")]
+extern static void FunctionWindows(int a);
+
+[DllImport("NativeLib", EntryPoint = "Function")]
+extern static void FunctionUnix(nint a);
+
+// Usage
+if (IsWindows)
+{
+    FunctionWindows(a);
+}
+else
+{
+    FunctionUnix(a);
+}
+```
+
+In .NET 6 a new type has been introduced to help ameliorate this issue - [`CLong` and `CULong`](https://github.com/dotnet/runtime/issues/13788). In .NET 6 the following is now possible. The example is for `CLong`, but you can use `CULong` to abstract C's `unsigned long` in a similar way.
```

Comment:
I would put the .NET 6+ solution first. My general rule for this sort of documentation is to optimize for future and put the simple modern way of doing stuff first, so that the people that are not the latest version of the platform do not need to read the whole history.